pkg/controller/mpi_job_controller_test.go

```diff
@@ -504,6 +509,21 @@ func TestDoNothingWithInvalidMPIJob(t *testing.T) {
 	f.run(getKey(mpiJob, t))
 }
 
+func TestDoNothingWithMPIJobManagedExternally(t *testing.T) {
```

Comment:
Also validate that a nil value the ManagedBy field leads to processing.

This is important for jobs that were created before the ManagedBy field existed

Reply:
Sure, we could add extra test case, but all other tests use mpiJob returned from `newMPIJob`, where managedBy is nil by default.
Although I could enhance that by adding it explicitly:
```go
func newMPIJobCommon(name string, startTime, completionTime *metav1.Time) *kubeflow.MPIJob {
...
	mpiJob := &kubeflow.MPIJob{
...
		Spec: kubeflow.MPIJobSpec{
			RunPolicy: kubeflow.RunPolicy{
				CleanPodPolicy: &cleanPodPolicyAll,
				ManagedBy: nil,
			},
```
wdyt?

Reply:
No, that's not necessary. Thanks for checking.